plugins/woocommerce-admin/client/products/sections/pricing-section.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Link, useFormContext } from '@woocommerce/components';
+import { Product, SETTINGS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { useContext } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import interpolateComponents from '@automattic/interpolate-components';
+import {
+	// @ts-expect-error `__experimentalInputControl` does exist.
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './pricing-section.scss';
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import { getInputControlProps } from './utils';
+import { ADMIN_URL } from '../../utils/admin-settings';
+import { CurrencyContext } from '../../lib/currency-context';
+import {
+	NUMBERS_AND_DECIMAL_SEPARATOR,
+	ONLY_ONE_DECIMAL_SEPARATOR,
+} from '../constants';
+
+export const PricingSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const { isResolving: isTaxSettingsResolving, taxSettings } = useSelect(
+		( select ) => {
+			const { getSettings, hasFinishedResolution } =
+				select( SETTINGS_STORE_NAME );
+			return {
+				isResolving: ! hasFinishedResolution( 'getSettings', [
+					'tax',
+				] ),
+				taxSettings: getSettings( 'tax' ).tax || {},
+			};
+		}
+	);
+	const pricesIncludeTax =
+		taxSettings.woocommerce_prices_include_tax === 'yes';
+	const context = useContext( CurrencyContext );
+	const { getCurrencyConfig } = context;
+	const { decimalSeparator } = getCurrencyConfig();
+	const priceValidation = ( value: string, name: string ) => {
```

Comment:
It would be awesome to extract most of this logic into a separate function so that it's easier to test and this function becomes:

```js
const sanitizedValue = sanitizePrice( value );
setValue( name, sanitizedValue );
```